features/accommodations/preview/BookingCard.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, type FC, useEffect, useMemo } from "react";
+import { type PriceDiff } from "../AccommodationModels";
+import moment from "moment";
+import IntegerInput from "../../../core/components/IntegerInput";
+import { Button } from "../../../core/components/Button";
+import { matches, parseCronExpression } from "../../../core/utils/cron";
+
+type BookingCardProps = {
+  price: number;
+  isPerGuest: boolean;
+  priceDiffs: PriceDiff[];
+  checkIn?: Date;
+  checkOut?: Date;
+  minGuests: number;
+  maxGuests: number;
+};
+
+type PriceInfo = {
+  items: { dateStr: string; price: number }[];
+  days: number;
+  total: number;
+};
+
+export const BookingCard: FC<BookingCardProps> = ({
+  price,
+  isPerGuest,
+  priceDiffs,
+  checkIn,
+  checkOut,
+  minGuests,
+  maxGuests,
+}) => {
+  const [priceInfo, setPriceInfo] = useState<PriceInfo>();
+  const [showDetails, setShowDetails] = useState(false);
+  const [guests, setGuests] = useState(1);
+  const priceDiffData = useMemo(
+    () =>
+      priceDiffs.map((pd) => ({
+        numbers: parseCronExpression(pd.when),
+        percentage: pd.percentage,
+      })),
+    [priceDiffs]
+  );
+
+  useEffect(() => {
+    if (!checkIn || !checkOut) {
+      setPriceInfo(undefined);
+      return;
+    }
+    let date = new Date(checkIn);
```

Comment:
This looks like calculating price logic. Can you extract this to some util method so we can reuse it if needed.